File: framework/languages/java/e2e/RuntimeMonitoring/Shared/src/main/java/systems/zlink/e2e/runtimemonitoring/shared/Contracts.java
package systems.zlink.e2e.runtimemonitoring.shared;

import java.util.List;

public final class Contracts {
    public static final String CHANNEL = "monitoring.api";
    public static final String HANDSHAKE_CHANNEL = "monitoring.handshake";
    public static final String SPOT_MESH = "monitoring.spot.mesh";
    public static final String SPOT_CHANNEL = "monitoring.spot.runtime";
    public static final String MONITORING_SPOT_TYPE = "monitoring-room";
    public static final String TRIGGERED_MONITORING_SPOT_TYPE =
        "monitoring-triggered";
    public static final String HANDLER_GROUP = "monitoring";
    public static final String LOCATION_SOURCE = "ops-locations";

    private Contracts() {
    }

    public record WorkReq(String value) {
    }

    public record WorkRes(String value, String providerRid) {
    }

    public record SpotSubjectProbe(String value) {
    }

    public record EvidenceEntry(String surface, String sourceName, String event, String detail) {
    }

    public record EvidenceSnapshot(List<EvidenceEntry> entries) {
    }

    public record ObserverIsolationStatus(
        boolean started,
        long normalEventCount,
        long normalLatestSequence,
        long slowLatestSequence,
        boolean slowFailed) {
    }

    public record RuntimePeer(
        String nodeRid,
        String state,
        String unavailableReason) {
    }

    public record RuntimeChannel(
        String channelName,
        boolean ready,
        int readyTargetCount) {
    }

    public record RuntimeSnapshot(
        String meshName,
        String state,
        boolean ready,
        int readyPeerCount,
        long sequence,
        String observedAt,
        List<RuntimePeer> peers,
        List<RuntimeChannel> channels,
        boolean placementAvailable,
        int activeActorCount,
        int activeSpotCount,
        String placementUnavailableReason,
        String hostState) {
    }
}

File: framework/languages/java/e2e-kotlin/RuntimeMonitoring/Shared/src/main/kotlin/systems/zlink/e2e/kotlin/runtimemonitoring/Contracts.kt
package systems.zlink.e2e.kotlin.runtimemonitoring

class Contracts private constructor() {
    companion object {
        const val CHANNEL: String = "monitoring.api"
        const val HANDSHAKE_CHANNEL: String = "monitoring.handshake"
        const val SPOT_MESH: String = "monitoring.spot.mesh"
        const val SPOT_CHANNEL: String = "monitoring.spot.runtime"
        const val HANDLER_GROUP: String = "monitoring"
    const val LOCATION_SOURCE: String = "ops-locations"
    const val ACTOR_TYPE: String = "monitoring-actor"
    }

    @JvmRecord
    data class WorkReq(val value: String)

    @JvmRecord
    data class WorkRes(
        val value: String,
        val providerRid: String,
    )

    @JvmRecord
    data class EvidenceEntry(
        val surface: String,
        val sourceName: String,
        val event: String,
        val detail: String,
    )

    @JvmRecord
    data class EvidenceSnapshot(val entries: List<EvidenceEntry>)

    @JvmRecord
    data class RuntimeSnapshot(
        val meshName: String,
        val state: String,
        val ready: Boolean,
        val readyPeerCount: Int,
        val sequence: Long,
        val peers: List<RuntimePeer>,
        val channels: List<RuntimeChannel>,
        val placementAvailable: Boolean,
        val activeActorCount: Int,
        val activeSpotCount: Int,
        val placementUnavailableReason: String,
    )

    @JvmRecord
    data class RuntimePeer(
        val nodeRid: String,
        val state: String,
        val unavailableReason: String,
    )

    @JvmRecord
    data class RuntimeChannel(
        val channelName: String,
        val ready: Boolean,
        val readyTargetCount: Int,
    )

    @JvmRecord
    data class ObserverStatus(
        val started: Boolean,
        val latestSequence: Long,
        val latestReady: Boolean,
        val latestReadyPeerCount: Int,
        val error: String,
    )
}
